### ISSUE TITLE
High Embedding Similarity Between Biological and Random Sequences

### ISSUE THREAD

**Issue by xzx0554** [ArcInstitute/evo2#97]
Hello, thank you so much for publicly sharing such outstanding work. Recently, while testing the embeddings of the model you released, we observed that after average or max pooling the embeddings from the final layer, the cosine similarity between biological sequences and random sequences was 1. In our tests, the similarity between biological embeddings was as high as 97% in most cases. Meanwhile, we modified your model by removing the `umemd` and adding an MLP classification head. We trained it on over a million sequences for binary classification under both causal and bidirectional attention mechanisms, but the loss consistently failed to decrease. Has your organization conducted related experiments? Could you provide some guidance or suggestions?

**Comment by xzx0554** [comment]
In the example sequences provided by Evo2, the test length is 500, while in my fine-tuned version (https://github.com/xzx0554/EVO2-Virus), the test length is 105. Both models can generate valid predicted sequences when tested, but the pooled embeddings show high similarity to those of random sequences.

**Comment by xzx0554** [comment]
thank u so much for your reply ,Your suggestions helped us modify our own code, and it has shown excellent results!

**Comment by garykbrixi** [comment]
Hi @ChIPseq, for exon task it was specifically `blocks.26` of Evo 2 7b base
